### ISSUE TITLE
With testing-farm based test execution, output is lost making tests end up as error

### ISSUE THREAD

**Issue by vhumpa** [teemtee/tmt#2381]
Hello,

Based on yesterday's query at the testing-farm open hours meeting, I was advised to file this issue for tmt.

**Description:**
When executing our tests through testing-farm, we lose output in the middle of our test script (`runtest.sh`) after the following command: 
```
sudo -u test qecore-headless --session-type xorg --keep-max 'behave -t browse_tabs -k -f html-pretty -o /tmp/report_gnome-system-monitor_005_browse_tabs.html -f plain'
```

Refer to this link for the artifacts: http://artifacts.osci.redhat.com/testing-farm/01b710c5-043a-4940-aa99-5418b3e8a24a/

The tests marked as errors always lose their output after this command is executed. Based on tmt's `log.txt`, the return code is 255. In contrast, any passing result displays the output as expected, all the way to the final reporting, and exits with either 1 or 0.

It can also be seen with passing logs that we store debug information from the test script, which runs under qecore-headless. For instance, `/tmp/dogtail-test/logs/behave_20231003-054627_debug` contains all the dogtail interaction details even for every ERR test. Thus, the tests appear to have run and mostly conclude normally; it's just that the output seems to disappear for some reason.

Also, the completion time, as seen in the log, seems to be as expected, 30 seconds as in this example:
```
09:45:46                 out: + sudo -u test qecore-headless --session-type xorg --keep-max 'behave -t startViaCommand -k -f html-pretty -o /tmp/report_gnome-system-monitor_001_startViaCommand.html -f plain'
09:46:17                 out: Shared connection to 10.31.14.90 closed.
```

**Reproduction:** 
For the `gnome-system-monitor` example, the issue is consistently reproducible using testing-farm on redhat ranch by executing the following (it's possible to add: --environment TF_HACK_RESERVE_SECONDS=3600 etc. to make system "reserve" for an hour after the tests have completed):
```
testing-farm request --compose RHEL-9.3.0-Nightly --git-url https://gitlab.cee.redhat.com/desktopqe/gnome-system-monitor.git --git-ref devel/el9 --plan /plan --test-filter "tag:gate"
```

**Additional Information:**
- With the provided example, the same tests consistently result in an error, and the ones that pass always do so. This is peculiar because, at the `runtest.sh`, headless script, or even the behave command level, there's no difference in what the tests do. They just interact with different UI elements in the resulting graphical session.
- When I manually SSH into the reserved system after the tests have completed, executing the tests directly with `./runtest.sh testname` works without any issues.
- A brief description of the `qecore-headless` process: it initiates the graphical session by triggering "systemctl start gdm.service" with autologin enabled for the 'test' user. Subsequently, it waits for a local binary to start (`gnome-shell`), "steals" the environment from it (from `/proc/`), and runs the provided script in that session, particularly the `behave` command. Using the `gnome-shell` process environment, `behave` behaves as if it's running in that session, even though it's executed externally. This effectively simulates the environment needed for Dogtail-based tests (with `behave` as the test framework glue) to operate.
- We've tried various alternatives such as using different shells, capturing both stderr and stdout to a file (which also results in an empty file), and swapping the `sudo` command with "su -c". Sadly, none have proven successful.
- The TF team suspects this issue is related to `tmt`, particularly its SSH-based execution and the specific command structure (i.e., `sudo` running tasks as a test user which then runs additional tasks). But that was a quick guess and the issue could of course lie with testing-farm or elsewhere.

I'm available to assist with any additional information or testing if required, both online and in person. Please let me know if there are any resources under NAT that are needed, given that this was executed on RH internal infrastructure and this is a GitHub issue.

**Comment by lukaszachy** [comment]
Good news - can be reproduced on 1mt images. Bad news - no idea about the cause for now.

**Comment by lukaszachy** [comment]
That is the plan once time allows.

**Comment by thrix** [comment]
Trying also to get answer at least to what I was interested in

**Comment by thrix** [comment]
hmm, the reproducer passed just fine in Openstack run via the 1mt tmt plugin, trying to rerun via Testing Farm forcing pool to Openstack:

http://artifacts.osci.redhat.com/testing-farm/0b749f60-a283-4fc4-b1b3-07e0702ebfab

**Comment by happz** [comment]
> hmm, the reproducer passed just fine in Openstack run via the 1mt tmt plugin, trying to rerun via Testing Farm forcing pool to Openstack:
> 
> http://artifacts.osci.redhat.com/testing-farm/0b749f60-a283-4fc4-b1b3-07e0702ebfab

Run it multiple times. My experience with `virtual` suggested series: five runs went well, then three failed, thene several runs went well again, then a series of failed ones, and so on.

**Comment by thrix** [comment]
Agreed, it is erroring out a lot:

http://artifacts.osci.redhat.com/testing-farm/0b749f60-a283-4fc4-b1b3-07e0702ebfab/

Needs another day

**Comment by vhumpa** [comment]
Hi! Thanks for looking into this! Sorry about the delay after tests and before the pipeline fnishing, the original reproducer had the TF_HACK_RESERVE_SECONDS set to 10000 which causes the system to "reserve" at the finish step for several hours. Used that for debugging.

**Comment by thrix** [comment]
@vhumpa sorry, we have been busy, I will resume work on this

**Comment by lukaszachy** [comment]
I've started to look into this again and this is suspicious:
```
/var/log/secure

Nov 28 06:53:40 default-0 sudo[72845]:    root : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=test ; COMMAND=/bin/qecore-headless --session-type xorg --keep-max behave -t browse_tabs -k -f html-pretty -o /tmp/report_gnome-system-monitor_005_browse_tabs.html -f plain
Nov 28 06:53:40 default-0 sudo[72845]: pam_unix(sudo:session): session opened for user test(uid=1000) by root(uid=0)
Nov 28 06:53:40 default-0 sudo[72851]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/mv -f /tmp/custom.conf /etc/gdm/custom.conf
Nov 28 06:53:40 default-0 sudo[72851]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:40 default-0 sudo[72851]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:40 default-0 sudo[72854]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/rm -f /tmp/custom.conf
Nov 28 06:53:40 default-0 sudo[72854]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:40 default-0 sudo[72854]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:40 default-0 sudo[72865]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/mv -f /tmp/org.gnome.SettingsDaemon.A11ySettings.service /usr/lib/systemd/user/org.gnome.SettingsDaemon.A11ySettings.service
Nov 28 06:53:40 default-0 sudo[72865]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:40 default-0 sudo[72865]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:40 default-0 sudo[72868]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/rm -f /tmp/org.gnome.SettingsDaemon.A11ySettings.service
Nov 28 06:53:40 default-0 sudo[72868]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:40 default-0 sudo[72868]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:40 default-0 sudo[72872]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/rm -f /tmp/org.gnome.Shell@wayland.service
Nov 28 06:53:40 default-0 sudo[72872]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:40 default-0 sudo[72872]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:40 default-0 sudo[72877]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/systemctl start gdm
Nov 28 06:53:40 default-0 sudo[72877]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:40 default-0 sudo[72877]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:40 default-0 systemd[72890]: pam_unix(systemd-user:session): session opened for user test(uid=1000) by (uid=0)
Nov 28 06:53:40 default-0 gdm-autologin][72884]: pam_unix(gdm-autologin:session): session opened for user test(uid=1000) by (uid=0)
Nov 28 06:53:40 default-0 gdm-autologin][72884]: gkr-pam: gnome-keyring-daemon started properly
Nov 28 06:53:41 default-0 polkitd[41673]: Registered Authentication Agent for unix-session:54 (system bus name :1.2008 [/usr/bin/gnome-shell], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8)
Nov 28 06:53:44 default-0 sudo[73554]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=test ; COMMAND=/bin/gsettings get org.gnome.desktop.peripherals.keyboard delay
Nov 28 06:53:44 default-0 sudo[73554]: pam_unix(sudo:session): session opened for user test(uid=1000) by root(uid=1000)
Nov 28 06:53:44 default-0 sudo[73554]: pam_unix(sudo:session): session closed for user test
Nov 28 06:53:44 default-0 sudo[73559]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=test ; COMMAND=/bin/gsettings get org.gnome.desktop.peripherals.keyboard repeat
Nov 28 06:53:44 default-0 sudo[73559]: pam_unix(sudo:session): session opened for user test(uid=1000) by root(uid=1000)
Nov 28 06:53:44 default-0 sudo[73559]: pam_unix(sudo:session): session closed for user test
Nov 28 06:53:44 default-0 sudo[73564]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=test ; COMMAND=/bin/gsettings get org.gnome.desktop.peripherals.keyboard repeat-interval
Nov 28 06:53:44 default-0 sudo[73564]: pam_unix(sudo:session): session opened for user test(uid=1000) by root(uid=1000)
Nov 28 06:53:44 default-0 sudo[73564]: pam_unix(sudo:session): session closed for user test
Nov 28 06:53:44 default-0 sudo[73570]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/rm -rf /tmp/qecore_logger.log
Nov 28 06:53:44 default-0 sudo[73570]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:44 default-0 sudo[73570]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:44 default-0 sudo[73609]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/journalctl --lines=0 --show-cursor
Nov 28 06:53:44 default-0 sudo[73609]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:44 default-0 sudo[73609]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:46 default-0 sudo[73632]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/rm -rf /home/test/Videos/Screencast*
Nov 28 06:53:46 default-0 sudo[73632]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:46 default-0 sudo[73632]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:56 default-0 sudo[73687]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/kill -9 73635
Nov 28 06:53:56 default-0 sudo[73687]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:56 default-0 sudo[73687]: pam_unix(sudo:session): session closed for user root
Nov 28 06:53:59 default-0 sudo[73717]:    test : TTY=pts/1 ; PWD=/var/tmp/tmt/run-221/plan/discover/default-0/tests ; USER=root ; COMMAND=/bin/kill -9 72712
Nov 28 06:53:59 default-0 sudo[73717]: pam_unix(sudo:session): session opened for user root(uid=0) by root(uid=1000)
Nov 28 06:53:59 default-0 sudo[72845]: pam_unix(sudo:session): session closed for user test
Nov 28 06:53:59 default-0 sudo[73717]: pam_unix(sudo:session): session closed for user root
```

See the `COMMAND=/bin/kill -9 72712` line.

I've done `ps aux` just before the test started and it is the 'ssh test session itself'
```
root       72712  0.0  0.0 222672  3584 pts/1    Ss+  06:51   0:00 bash -c export TEST=gnome-system-monitor_005_browse_tabs; export TMT_PLAN_DATA=/var/tmp/tmt/run-221/plan/data; export TMT_PLAN_ENVIRONMENT_FILE=/var/tmp/tmt/run-221/plan/data/variables.env; export TMT_TREE=/var/tmp/tmt/run-221/plan/tree; export TMT_VERSION=1.29.0; export TMT_TEST_PIDFILE=/var/tmp/tmt-test.pid; export TMT_TEST_PIDFILE_LOCK=/var/tmp/tmt-test.pid.lock; export TMT_TEST_NAME=/tests/browse_tabs; export TMT_TEST_DATA=/var/tmp/tmt/run-221/plan/execute/data/guest/default-0/tests/browse_tabs-3/data; export TMT_TEST_SERIAL_NUMBER=3; export TMT_TEST_METADATA=/var/tmp/tmt/run-221/plan/execute/data/guest/default-0/tests/browse_tabs-3/metadata.yaml; export TMT_REBOOT_REQUEST=/var/tmp/tmt/run-221/plan/execute/data/guest/default-0/tests/browse_tabs-3/data/reboot-request; export TMT_REBOOT_COUNT=0; export REBOOTCOUNT=0; export RSTRNT_REBOOTCOUNT=0; export TMT_TOPOLOGY_YAML=/var/tmp/tmt/run-221/plan/execute/data/guest/default-0/tests/browse_tabs-3/tmt-test-topology.yaml; export TMT_TOPOLOGY_BASH=/var/tmp/tmt/run-221/plan/execute/data/guest/default-0/tests/browse_tabs-3/tmt-test-topology.sh; cd /var/tmp/tmt/run-221/plan/discover/default-0/tests; [ ! -z "$TMT_DEBUG" ] && set -x  test_pid="$$";  mkdir -p "$(dirname $TMT_TEST_PIDFILE_LOCK)"       set -o pipefail;  flock "$TMT_TEST_PIDFILE_LOCK" -c "echo '${test_pid} ${TMT_REBOOT_REQUEST}' > ${TMT_TEST_PIDFILE}" || exit 122;     ./tmt-test-wrapper.sh.default-0 </dev/null |& cat;     _exit_code="$?";  flock "$TMT_TEST_PIDFILE_LOCK" -c "rm -f ${TMT_TEST_PIDFILE}" || exit 123;  exit $_exit_code;
```

Any idea why it runs? I've found the `kill -9` only in `features/environment.py`

**Comment by lukaszachy** [comment]
Doesn't explain why there is no output though.

**Comment by thrix** [comment]
@lukaszachy did you try to reproduce on 1minutetip? I tried with a test, but it seems to work :(

```
                cmd: ./runtest.sh show_process_properties
                out: +./runtest.sh:4> bash env
                out: /usr/bin/env: /usr/bin/env: cannot execute binary file
                out: +./runtest.sh:6> whereis rhts-report-result
                out: rhts-report-result: /usr/local/bin/rhts-report-result
                out: +./runtest.sh:9> [ ! -e /tmp/qecore_setup_done ']'
                out: +./runtest.sh:25> sudo service gdm stop
                out: Redirecting to /bin/systemctl stop gdm.service
                out: +./runtest.sh:26> env 'GIT_SSL_NO_VERIFY=true' git submodule update --init --recursive
                out: ./runtest.sh:28: no matches found: /home/test/Videos/*
                out: +./runtest.sh:31> sudo -u test qecore-headless --session-type xorg --keep-max 'behave -t show_process_properties -k -f html-pretty -o /tmp/report_gnome-system-monitor_034_show_process_properties.html -f plain'
                out: ConfigError: No steps directory in '/var/home/mvadkert/.local/share/tmt/run-627/plan/discover/default-0/tests/features'
                out: headless: Starting the timer for the headless script.
                out: headless: Enabling G_MESSAGES_DEBUG for accessibility.
                out: headless: Starting Display Manager
                out: headless: Setting environment variable TERM as 'xterm'
                out: headless: Running 'xorg' with desktop 'gnome'
                out: headless: Setting DISPLAY variable to ':0'.
                out: headless: Started the script with PID 45962
                out: headless: The user script finished with return code 1
                out: headless: The headless script ran for 6.84 seconds.
                out: +./runtest.sh:31> rc=1 
                out: +./runtest.sh:33> sleep 2
                out: +./runtest.sh:35> RESULT=FAIL 
                out: +./runtest.sh:36> [ 1 -eq 0 ']'
                out: +./runtest.sh:40> tmt-file-submit -l /tmp/report_gnome-system-monitor_034_show_process_properties.html
                out: cp: cannot stat '/tmp/report_gnome-system-monitor_034_show_process_properties.html': No such file or directory
                out: +./runtest.sh:41> tmt-report-result gnome-system-monitor_034_show_process_properties FAIL
                out: +./runtest.sh:45> exit 1
                out: Shared connection to 10.0.185.189 closed.
        Command returned '1'.
        Append to file '/var/home/mvadkert/.local/share/tmt/run-627/plan/execute/data/guest/default-0/tests/show_process_properties-1/output.txt'.
        Copy '/var/home/mvadkert/.local/share/tmt/run-627/plan/execute/data/guest/default-0/tests/show_process_properties-1' from the guest to '/'.
        Run command: rsync -s -R -r -z --links --safe-links --protect-args -e 'ssh -oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oServerAliveInterval=60 -oServerAliveCountMax=5 -oIdentitiesOnly=yes -i /var/home/mvadkert/.local/share/tmt/run-627/plan/provision/default-0/1minutetip -i /var/home/mvadkert/.local/share/tmt/run-627/plan/provision/default-0/1minutetip-ecdsa -S/run/user/1000/tmt/tmpvbh7mh0m' root@10.0.185.189:/var/home/mvadkert/.local/share/tmt/run-627/plan/execute/data/guest/default-0/tests/show_process_properties-1 /
        Extract results of '/tests/show_process_properties'.
        tmt-report-results file '/var/home/mvadkert/.local/share/tmt/run-627/plan/execute/data/guest/default-0/tests/show_process_properties-1/data/restraint-result detected.
                00:00:09 fail /tests/show_process_properties (on default-0) [1/1]
```

**Comment by thrix** [comment]
Yeah, cannot reproduce on 1mt, tests just failed, I used the reproducer steps from:

http://artifacts.osci.redhat.com/testing-farm/0b749f60-a283-4fc4-b1b3-07e0702ebfab/

```
❯ tmt --root . run --last report
/var/home/mvadkert/.local/share/tmt/run-628

/plan
    report
        how: display
        summary: 47 tests failed

total: 47 tests failed
```

**Comment by thrix** [comment]
ok, I am able to still reproduce via TF, but I was not able to reproduce manually from the worker, will try again tmrw

**Comment by lukaszachy** [comment]
I use 1mt image  but with testcloud

**Comment by vhumpa** [comment]
Thank you for looking over at this, let me know if there's anything specific to help!

**Comment by thrix** [comment]
I have reproduced it manually finally \o/:

```
[root@bdfbaccb-8de6-4071-a708-f722395eba5f work-plan7jj_xvrr]# ssh -oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oServerAliveInterval=60 -oServerAliveCountMax=5 -oIdentitiesOnly=yes -p22 -i /etc/citool.d/id_rsa_artemis -S/run/user/0/tmt/tmphm4tur51 -tt root@10.0.177.11 'export TEST=gnome-system-monitor_034_show_process_properties; export TF_HACK_RESERVE_SECONDS=10000; export TESTING_FARM_REQUEST_ID=00a51bad-4f7a-404f-8ce6-552a4d2ecee6; export TESTING_FARM_TEST_TYPE=fmf; export TESTING_FARM_GIT_URL=https://gitlab.cee.redhat.com/desktopqe/gnome-system-monitor.git; export TESTING_FARM_GIT_REF=devel/el9; export TMT_PLAN_DATA=/var/ARTIFACTS/work-plan7jj_xvrr/plan/data; export TMT_PLAN_ENVIRONMENT_FILE=/var/ARTIFACTS/work-plan7jj_xvrr/plan/data/variables.env; export TMT_TREE=/var/ARTIFACTS/work-plan7jj_xvrr/plan/tree; export TMT_VERSION=1.29.0; export TMT_TEST_PIDFILE=/var/tmp/tmt-test.pid; export TMT_TEST_PIDFILE_LOCK=/var/tmp/tmt-test.pid.lock; export TMT_TEST_NAME=/tests/show_process_properties; export TMT_TEST_DATA=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/data; export TMT_TEST_SERIAL_NUMBER=7; export TMT_TEST_METADATA=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/metadata.yaml; export TMT_REBOOT_REQUEST=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/data/reboot-request; export TMT_REBOOT_COUNT=0; export REBOOTCOUNT=0; export RSTRNT_REBOOTCOUNT=0; export TMT_TOPOLOGY_YAML=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/tmt-test-topology.yaml; export TMT_TOPOLOGY_BASH=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/tmt-test-topology.sh; cd /var/ARTIFACTS/work-plan7jj_xvrr/plan/discover/default-0/tests; [ ! -z "$TMT_DEBUG" ] && set -x

test_pid="$$";

mkdir -p "$(dirname $TMT_TEST_PIDFILE_LOCK)"


    set -o pipefail;

flock "$TMT_TEST_PIDFILE_LOCK" -c "echo '"'"'${test_pid} ${TMT_REBOOT_REQUEST}'"'"' > ${TMT_TEST_PIDFILE}" || exit 122;
    ./tmt-test-wrapper.sh.default-0 </dev/null |& cat;
    _exit_code="$?";

flock "$TMT_TEST_PIDFILE_LOCK" -c "rm -f ${TMT_TEST_PIDFILE}" || exit 123;

exit $_exit_code;'
Warning: Identity file /etc/citool.d/id_rsa_artemis not accessible: No such file or directory.
Warning: Permanently added '10.0.177.11' (ED25519) to the list of known hosts.
+./runtest.sh:4> bash env
/usr/bin/env: /usr/bin/env: cannot execute binary file
+./runtest.sh:6> whereis rhts-report-result
rhts-report-result: /usr/local/bin/rhts-report-result
+./runtest.sh:9> [ ! -e /tmp/qecore_setup_done ']'
+./runtest.sh:25> sudo service gdm stop
Redirecting to /bin/systemctl stop gdm.service
+./runtest.sh:26> env 'GIT_SSL_NO_VERIFY=true' git submodule update --init --recursive
+./runtest.sh:28> rm -rf /home/test/Videos/Screencast.webm
+./runtest.sh:31> sudo -u test qecore-headless --session-type xorg --keep-max 'behave -t show_process_properties -k -f html-pretty -o /tmp/report_gnome-system-monitor_034_show_process_properties.html -f plain'
Killed
```

It has something to do with the `tmt-test-wrapper.sh`, when I run the test directly, it works:

```
[root@bdfbaccb-8de6-4071-a708-f722395eba5f work-plan7jj_xvrr]# ssh -oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oServerAliveInterval=60 -oServerAliveCountMax=5 -oIdentitiesOnly=yes -p22 -i /etc/citool.d/id_rsa_artemis -S/run/user/0/tmt/tmphm4tur51 -tt root@10.0.177.11 'export TEST=gnome-system-monitor_034_show_process_properties; export TF_HACK_RESERVE_SECONDS=10000; export TESTING_FARM_REQUEST_ID=00a51bad-4f7a-404f-8ce6-552a4d2ecee6; export TESTING_FARM_TEST_TYPE=fmf; export TESTING_FARM_GIT_URL=https://gitlab.cee.redhat.com/desktopqe/gnome-system-monitor.git; export TESTING_FARM_GIT_REF=devel/el9; export TMT_PLAN_DATA=/var/ARTIFACTS/work-plan7jj_xvrr/plan/data; export TMT_PLAN_ENVIRONMENT_FILE=/var/ARTIFACTS/work-plan7jj_xvrr/plan/data/variables.env; export TMT_TREE=/var/ARTIFACTS/work-plan7jj_xvrr/plan/tree; export TMT_VERSION=1.29.0; export TMT_TEST_PIDFILE=/var/tmp/tmt-test.pid; export TMT_TEST_PIDFILE_LOCK=/var/tmp/tmt-test.pid.lock; export TMT_TEST_NAME=/tests/show_process_properties; export TMT_TEST_DATA=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/data; export TMT_TEST_SERIAL_NUMBER=7; export TMT_TEST_METADATA=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/metadata.yaml; export TMT_REBOOT_REQUEST=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/data/reboot-request; export TMT_REBOOT_COUNT=0; export REBOOTCOUNT=0; export RSTRNT_REBOOTCOUNT=0; export TMT_TOPOLOGY_YAML=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/tmt-test-topology.yaml; export TMT_TOPOLOGY_BASH=/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/tmt-test-topology.sh; cd /var/ARTIFACTS/work-plan7jj_xvrr/plan/discover/default-0/tests; set -eo pipefail; ./runtest.sh show_process_properties'
Warning: Identity file /etc/citool.d/id_rsa_artemis not accessible: No such file or directory.
Warning: Permanently added '10.0.177.11' (ED25519) to the list of known hosts.
+./runtest.sh:4> bash env
/usr/bin/env: /usr/bin/env: cannot execute binary file
+./runtest.sh:6> whereis rhts-report-result
rhts-report-result: /usr/local/bin/rhts-report-result
+./runtest.sh:9> [ ! -e /tmp/qecore_setup_done ']'
+./runtest.sh:25> sudo service gdm stop
Redirecting to /bin/systemctl stop gdm.service
+./runtest.sh:26> env 'GIT_SSL_NO_VERIFY=true' git submodule update --init --recursive
./runtest.sh:28: no matches found: /home/test/Videos/*
+./runtest.sh:31> sudo -u test qecore-headless --session-type xorg --keep-max 'behave -t show_process_properties -k -f html-pretty -o /tmp/report_gnome-system-monitor_034_show_process_properties.html -f plain'
headless: Starting the timer for the headless script.
headless: Enabling G_MESSAGES_DEBUG for accessibility.
headless: Starting Display Manager
headless: Setting environment variable TERM as 'xterm'
headless: Running 'xorg' with desktop 'gnome'
headless: Setting DISPLAY variable to ':0'.
headless: Started the script with PID 51012
Feature: Processes tests
Creating logfile at /tmp/dogtail-test/logs/behave_20231129-134219_debug ...
Clicking on [toggle button | ]
Mouse button 1 click at (60.0,16.0)

  Scenario: Show properties of a process
    * Start gnome-system-monitor ... passed in 2.353s
    * Go to a tab "Processes" ... passed in 1.007s
    * Choose to change priority of "gnome-system-monitor" to "Very Low" ... passed in 7.635s
    * Choose to "Properties" for "gnome-system-monitor" using context menu ... passed in 4.565s
    Then Verify you can see "gnome-system-monitor" within the properties ... passed in 0.279s
    Then Verify you can see "Running" within the properties ... passed in 0.236s
    Then Verify you can see "Very Low" within the properties ... passed in 0.234s
    Then Verify you can see "19" within the properties ... passed in 0.217s

1 feature passed, 0 failed, 1 skipped
1 scenario passed, 0 failed, 46 skipped
8 steps passed, 0 failed, 310 skipped, 0 undefined
Took 0m16.526s
headless: The user script finished with return code 0
headless: The headless script ran for 31.84 seconds.
+./runtest.sh:31> rc=0 
+./runtest.sh:33> sleep 2
+./runtest.sh:35> RESULT=FAIL 
+./runtest.sh:36> [ 0 -eq 0 ']'
+./runtest.sh:37> RESULT=PASS 
+./runtest.sh:40> tmt-file-submit -l /tmp/report_gnome-system-monitor_034_show_process_properties.html
File '/tmp/report_gnome-system-monitor_034_show_process_properties.html' stored to '/var/ARTIFACTS/work-plan7jj_xvrr/plan/execute/data/guest/default-0/tests/show_process_properties-7/data'.
+./runtest.sh:41> tmt-report-result gnome-system-monitor_034_show_process_properties PASS
+./runtest.sh:45> exit 0
Connection to 10.0.177.11 closed.
```

**Comment by thrix** [comment]
When I change 

```
    ./tmt-test-wrapper.sh.default-0 </dev/null |& cat;
```
to

```
    ./tmt-test-wrapper.sh.default-0 </dev/null ;
```

It works, a minimal change, so it has somehting to do with output redirection, or it does not like :cat: :dancers:

**Comment by thrix** [comment]
I need to check the code, to understand why is it there

**Comment by thrix** [comment]
The `cat` was introduced here:

https://github.com/teemtee/tmt/pull/1521#issuecomment-1306150717

For now changing it to `2>&1` resolves the issue for me, so I am going forward with this fix.

More discussion also here:

https://redhat-internal.slack.com/archives/C04LRPNDZ4Y/p1701283760514209

**Comment by thrix** [comment]
Ok, so after more discussion, we need to keep the default non-tty execution. This is what most of the tests expect.
We will introduce a new setting `tty: true` with which you will be able to run the test with tty available and that will resolve your issues. Unfortunately, there is no better way to resolve this problem.

It is not so different from what restraint supports:
https://restraint.readthedocs.io/en/latest/tasks.html?highlight=pty#use-pty
https://restraint.readthedocs.io/en/latest/tasks.html?highlight=pty#restraint-metadata-file

So we will be more backward compatible.

**Comment by thrix** [comment]
@vhumpa looks like we are getting this fixed in 1.30 \o/, I hope it will resolve your issue, you will need to add `tty: true` to your tests' metadata
